amr-wind/immersed_boundary/complex_terrain/hill_ops.H

```diff
@@ -57,23 +57,77 @@ struct InitDataOp<Hill>
                 const amrex::Real z0 = wdata.center_loc[2];
                 const amrex::Real h = wdata.height;
                 const amrex::Real b = wdata.half_width;
+                const amrex::Real fac = h * M_PI * M_PI / (b * b);
+
                 amrex::ParallelFor(
                     bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                        // Centroid of current cell
                         const amrex::Real x = problo[0] + (i + 0.5) * dx[0];
                         const amrex::Real y = problo[1] + (j + 0.5) * dx[1];
                         const amrex::Real z = problo[2] + (k + 0.5) * dx[2];
+
                         const amrex::Real phi_glob = phi(i, j, k);
-                        const amrex::Real dist = z - z0;
-                        const amrex::Real r = std::sqrt(
+                        const amrex::Real height = z - z0;
+                        amrex::Real r = std::sqrt(
                             (x - x0) * (x - x0) + (y - y0) * (y - y0));
+                        amrex::Real hillnear;
                         amrex::Real hill;
+
+                        // Initial guess
+                        amrex::Real rnear = 0.1 * b;
+                        amrex::Real f = 0.0;
+                        amrex::Real fp = 0.0;
+                        // Newton solve for the point on the IB closest to the
+                        // centroid of the current cell
+                        for (int n = 0; n < 10; n++) {
+                            amrex::Real cosr =
```

Comment:
Yep.  The only problem is that if there are inflections in the boundary (such as this particular hill shape), multiple local minima show up, and it can converge to the wrong one.  In this case, I put a comparison of the newton solve with the distance straight down (z-direction), and used the minimum of the two, and that was good enough (it produced the correct spatial distribution of phi).  But for arbitrary terrain, with perhaps many inflections, this approach won't be general enough.  Maybe AMRex already has something for finding minimum distance to boundaries.  Otherwise, I think there's a way to solve an elliptic PDE for wall distance.